splink/linker.py

```diff
@@ -168,17 +201,28 @@ def __init__(
         self._validate_input_dfs()
         self._em_training_sessions = []
 
-        self._names_of_tables_created_by_splink: list = []
+        self._names_of_tables_created_by_splink: set = set()
+        self._intermediate_table_cache: dict = CacheDictWithLogging()
 
         self._find_new_matches_mode = False
         self._train_u_using_random_sample_mode = False
         self._compare_two_records_mode = False
         self._self_link_mode = False
+        self._analyse_blocking_mode = False
 
         self._output_schema = ""
 
         self.debug_mode = False
 
+    @property
+    def _cache_uid(self):
+        # return self._settings_obj._cache_uuid
+        return self._cache_uuid
+
+    @_cache_uid.setter
+    def _cache_uid(self, value):
+        self._settings_obj._cache_uuid = value
+
```

Comment:
At the moment there is `linker._cache_uid`, `linker._cache_uuid`, and `linker._settings_obj._cache_uuid`. As this getter/setter refer to different objects, it is possible for the settings version to get out-of-sync with the others, although I can't currently find anywhere where this will cause an issue. Not sure if there is a semantic difference between all of these, but feels like it could be a source of issues

Reply:
Yeah - it's a good spot.

The underlying problem we're trying to work around is that it's possible to instantiate a linker without a settings object, which can occasionally be useful for e.g. exploratory analysis if the user doesn't want to write any settings

Probably we should probably just disallow instantiating a linker without a settings object, and then the getting could be simplified to `return self._settings_obj._cache_uuid`

There are other places in the codebase that would be simplified by this change as well.  It adds too much complexity to the codebase for a very minor (and probably seldom used) API improvement